app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -326,14 +326,28 @@ export class CloneRepository extends React.Component<
     this.updateAndValidatePath(newPath)
   }
 
-  private async doesPathExist(path: string) {
-    const exists = await pathExists(path)
-    // If the path changed while we were checking, we don't care anymore.
-    if (this.state.path !== path) {
-      return
-    }
+  private async validateEmptyFolder(path: string): Promise<null | Error> {
+    try {
+      const directoryFiles = await readdir(path)
+
+      // If the path has changed we no longer want the result
+      if (path !== this.state.path) {
+        return new Error('The path has been changed during check.')
+      }
+
+      if (directoryFiles.length === 0) {
+        return null
+      } else {
+        return new Error('The path folder contains files.')
```

Comment:
:art: this message should indicate that Git cannot clone to directory with files, if we're going to display it to the user